master/internal/api_experiment.go

```diff
@@ -2,11 +2,21 @@ package internal
 
 import (
 	"context"
+	"encoding/json"
 	"strings"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/model"
 )
 
+var badStateErr = status.Error(codes.FailedPrecondition, "experiment in incompatible state")
```

Comment:
blocking: We should have more descriptive errors, knowing what the state was that resulted in in the incompatible state.